libs/blocks/gnav/gnav-contextual-search.js

```diff
@@ -97,11 +98,17 @@ export default async function onSearchInput({ value, resultsEl, searchInputEl, a
 
   if (currentSearch === lastSearch) {
     if (!hits.length) {
-      const advancedLink = advancedSearchEl.querySelector('a');
-      const href = new URL(advancedLink.href);
-      href.searchParams.set('q', value);
-      advancedLink.href = href.toString();
-      resultsEl.replaceChildren(advancedSearchEl);
+      const noResults = await replaceKey('no-results', getConfig());
+      const emptyMessage = createTag('p', {}, noResults);
+      let emptyList = createTag('li', null, emptyMessage);
+      if (advancedSearchEl) {
+        const advancedLink = advancedSearchEl.querySelector('a');
+        const href = new URL(advancedLink.href);
+        href.searchParams.set('q', value);
+        advancedLink.href = href.toString();
```

Comment:
Would using the `href` property not achieve the same result?

Reply:
@overmyheadandbody are you suggesting something like this?
```
const advancedLink = advancedSearchEl.querySelector('a');
const { href } = advancedLink
advancedLink.href = `${href}?q=${value}`
```
In this case, we would be ignoring the existing params.